integration-test-groups/foundation/scheduler/src/test/java/org/apache/camel/quarkus/component/scheduler/it/SchedulerTest.java

```diff
@@ -28,12 +28,43 @@
 class SchedulerTest {
 
     @Test
-    public void test() throws Exception {
-        // wait until the scheduler has run and return a counter that is > 0
+    public void testInitialDelay() throws Exception {
         await().atMost(5, TimeUnit.SECONDS).until(() -> {
             String body = RestAssured.get("/scheduler/get").then().statusCode(200).extract().body().asString();
             return !body.equals("0");
         });
     }
 
+    @Test
+    public void testDelay() throws Exception {
+        await().atMost(2, TimeUnit.SECONDS).until(() -> {
+            String body = RestAssured.get("/scheduler/get-count").then().statusCode(200).extract().body().asString();
+            return Integer.parseInt(body) == 1;
+        });
+    }
+
+    @Test
+    public void testDelayWithFixed() throws Exception {
+        await().atMost(4, TimeUnit.SECONDS).until(() -> {
+            String body = RestAssured.get("/scheduler/get-count").then().statusCode(200).extract().body().asString();
+            return Integer.parseInt(body) > 1;
+        });
+    }
```

Comment:
It looks like you assume `testDelay()` to be executed before `testDelayWithFixed()` ? Maybe one test method would serve better?

```suggestion
    @Test
    public void testDelay() throws Exception {
        await().atMost(2, TimeUnit.SECONDS).until(() -> {
            String body = RestAssured.get("/scheduler/get-count").then().statusCode(200).extract().body().asString();
            return Integer.parseInt(body) > 1;
        });
        await().atMost(2, TimeUnit.SECONDS).until(() -> {
            String body = RestAssured.get("/scheduler/get-count").then().statusCode(200).extract().body().asString();
            return Integer.parseInt(body) > 2;
        });
    }
```